### ISSUE TITLE
Issue with thumbnail generation failure for extremely large images

### ISSUE THREAD

**Issue by guiyanakuang** [CrossPaste/crosspaste-desktop#1880]
Reported by @sunxiang0918 

In addition, I also found a BUG related to generating thumbnails.
When using skia to generate pictures (thumbnails), if the size of the picture is too large, the generation will fail and become an empty file with a small size.
On the CrossPaste interface, it is just a blank block.

<img width="1336" alt="image" src="https://github.com/user-attachments/assets/80368e98-e259-4140-8a11-11dbc0c04946">

I roughly looked for the reason, and what was said on the Internet may be related to CPU and memory:
https://groups.google.com/g/skia-discuss/c/A47-OtBOnKU
https://groups.google.com/g/skia-discuss/c/886WuVcQUGo
https://stackoverflow.com/questions/52206463/large-image-resizing-shows-blank

I tried it a little bit myself.
In the environment of my macmini (intel-i5 + 32g), the maximum size is 23170*23170 (sqrt(uint32/8)), and the program reports an error if it exceeds the size.
In my mbp (apple m2 + 16g) environment, I tested 128k*128k and it still works...
Therefore, I have not specifically tested what the boundaries and influencing factors are.

```
@Test
fun testCreateImage() {
    val width = 23170
    val height = 23170
    
    val surface = Surface.makeRasterN32Premul(width, height)
    val canvas = surface.canvas

    // fill gray background
    val paint = Paint().apply { color = Color.makeRGB(128, 128, 128) }
    canvas.drawRect(Rect.makeXYWH(0f, 0f, width.toFloat(), height.toFloat()), paint)

    val img = surface.makeImageSnapshot()

    val result = File("/Users/sun/Downloads/create1111.png").toOkioPath(true)
    
    img.encodeToData()?.bytes?.let {
        result.toNioPath().writeBytes(it)
    }
}
```

**Comment by guiyanakuang** [comment]
> 所以,有没有可能,当判断到图片尺寸比较大的时候, 就使用MacosApi.swift里面的方法 来生成缩略图?

Certainly, if using the native API on Mac is faster, I even support using MacosApi by default on Mac to implement this.

**Comment by guiyanakuang** [comment]
- [ ] #1907

**Comment by CompileFuture2024** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
- [ ] #1907

**Comment by sunxiang0918** [comment]
我对比了一下  `ImageMagick`和`libvips`.   在我的环境中, 相同的文件和相同的设置, libvips比ImageMagick快了3倍 (2.64s VS 8.29s)
<img width="1105" alt="image" src="https://github.com/user-attachments/assets/72882963-bf33-4a90-bfa5-5ccfe8d88bd9">

因此,我着重测试了一下通过JAVA调用 LibVips的方法和性能.
发现,使用JNA调用( [jlibvips](https://github.com/codecitizen/jlibvips)、 [jvips](https://github.com/criteo/JVips)) 非常的慢 (平均18s). 
但是如果使用JDK22的FFM(Foreign Function and Memory) API, 性能几乎和命令行是一样的. [vips-ffm](https://github.com/lopcode/vips-ffm)
供参考~

**Comment by guiyanakuang** [comment]
Exciting feature! It looks like JDK 22 can directly access off-heap memory, avoiding copying (which should be the main reason for slowness).

**Comment by guiyanakuang** [comment]
CrossPaste is currently bound to JBR on the desktop because JBR has fixed many clipboard-related bugs, while other open-source JDKs like OpenJDK have largely ignored these issues as they focus on the server side. The latest JBR supports JDK 21, so to use this new feature, you may need to wait for the next long-term support version.

**Comment by sunxiang0918** [comment]
是的, 所以可能ffm暂时还行不通. 
另一条路, 使用 `Runtime.getRuntime`,通过java调用shell命令, 避免图片的读取.

**Comment by CompileFuture2024** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
Yes, so maybe ffm won't work for the time being.
Another way is to use `Runtime.getRuntime` to call the shell command through java to avoid reading the image.
